common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -140,6 +140,52 @@ ICacheRecord save(@NonNull AccountRecord accountRecord,
         return result;
     }
 
+    // TODO Add unit test
+    @NonNull
+    List<ICacheRecord> saveAndLoadAggregatedAccountData(
+            @NonNull AccountRecord accountRecord,
+            @NonNull IdTokenRecord idTokenRecord,
+            @NonNull AccessTokenRecord accessTokenRecord) throws ClientException {
+        return mergeCacheRecordWithCorollaryAccounts(
+                save(accountRecord, idTokenRecord, accessTokenRecord)
+        );
+    }
+
+    @NonNull
+    private List<ICacheRecord> mergeCacheRecordWithCorollaryAccounts(
+            @NonNull final ICacheRecord savedCacheRecord) {
+        final List<ICacheRecord> result = new ArrayList<>();
+        // Whatever ICacheRecord you provide will _always_ be the first element in the result List.
+        result.add(savedCacheRecord);
+
+        final List<AccountRecord> corollaryAccounts = new ArrayList<>(
+                getCorollaryAccounts(
```

Comment:
Naming.... we probably need to simplify this....

Reply:
Took a pass at simplifying the language in 659d973a3ce00ffe8db67ed32d119eed05c36b60